packages/notification-services-controller/src/NotificationServicesController/types/on-chain-notification/on-chain-notification.ts

```diff
@@ -49,15 +50,23 @@ type ConvertToEnum<Kind> = {
  * 2. It ensures that the `data` field is the correct Notification data for this `type`
  * - The `Compute` utility merges the intersections (`&`) for a prettier type.
  */
-export type OnChainRawNotification = {
+type NormalizeNotification<
+  N extends Notification,
+  NotificationDataKinds extends string = NonNullable<N['data']>['kind'],
+> = {
   [K in NotificationDataKinds]: Compute<
-    Omit<Notification, 'data'> & {
+    Omit<N, 'data'> & {
       type: ConvertToEnum<K>;
-      data: Extract<Notification['data'], { kind: K }>;
+      data: Extract<N['data'], { kind: K }>;
     }
   >;
 }[NotificationDataKinds];
 
+export type OnChainRawNotification = Compute<
+  | NormalizeNotification<components['schemas']['WalletNotification']>
+  | NormalizeNotification<components['schemas']['Web3Notification']>
+>;
```

Comment:
Hehe, hope this type-fu was helpful.

I'll make a mental note on this to see if we can clean up the types. But it is what it is :P